HW1/P4/P4.py

```diff
@@ -0,0 +1,38 @@
+from P4_bfs import *
+from pyspark import SparkContext
+import itertools
+
+sc = SparkContext(appName="P4")
+
+# Format rows in dataset
+def format_line(line):
+	row = line.split(',')
+	book = row[-1]
+
+	# Join in case character has a comma in it
+	character = ','.join(row[:-1])
+
+	return (book, [character])
+
+if __name__ == '__main__':
+	# Create rdd of comic book characters
+	data = sc.textFile('source.csv')
+
+	# Construct graph
+	graph = data.map(format_line) \
+					.reduceByKey(lambda x, y: x+y) \
```

Comment:
Yeah.  That right there.  That was something that I did with an insane list comprehension with for...in...if... !=...